students/Dennis_Coffey/lesson03/mailroom.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Nov  8 19:30:19 2018
+
+@author: denni
+"""
+
+"""You work in the mail room at a local charity. Part of your job is to write incredibly boring, 
+repetitive emails thanking your donors for their generous gifts. 
+You are tired of doing this over and over again, so you’ve decided to let Python help 
+you out of a jam and do your work for you."""
+
+#The Program
+#Write a small command-line script called mailroom.py. This script should be executable. The script should accomplish the following goals:
+#It should have a data structure that holds a list of your donors and a history of the amounts they have donated. 
+#This structure should be populated at first with at least five donors, 
+#with between 1 and 3 donations each.
+#You can store that data structure in the global namespace.
+#The script should prompt the user (you) to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”)
+
+#Create list of donors
+donors = [['Dennis Coffey', [2500.00,400.00,1400.00]], 
+          ['Bill Gates', [120.00,650.00]], 
+          ['Ethan Coffey', [800.00,150.00,1100.00]], 
+          ['Paul Allen', [45000.00,9000.00]], 
+          ['Jeff Bezos', [3.00]]]
+
+
+#Sending a Thank You
+#If the user (you) selects ‘Send a Thank You’, prompt for a Full Name. 
+#If the user types ‘list’, show them a list of the donor names and re-prompt
+#If the user types a name not in the list, add that name to the data structure and use it.
+#If the user types a name in the list, use it.
+#Once a name has been selected, prompt for a donation amount.
+#Turn the amount into a number – it is OK at this point for the program to crash if someone types a bogus amount.
+#Once an amount has been given, add that amount to the donation history of the selected user.
+#Finally, use string formatting to compose an email thanking the donor for their generous donation. 
+#Print the email to the terminal and return to the original prompt.
+def send_thankyou():
+    #Loop if user selects list
+    full_name = 'list'
+    while full_name.lower() == 'list':
+        #Create prompt menu
+        full_name = input('Please input your Full Name\n'
+                          '\t or list if you would like to see a list of donors >> ')
+        #Create list of donor names
+        donor_names = []
+        for donor in donors:
+            donor_names.append(donor[0])
+            
+        #Check user input and perform appropriate action    
+        if full_name.lower() == 'list':
+            [print(f'{donor[0]}') for donor in donors]
```

Comment:
using `print` with list comprehensions isn't a suitable use case - `print()` returns None, so you are allocating N number of `None` values in a produced list.
```
>>> [print(x) for x in range(10)]
0
1
2
3
4
5
6
7
8
9
[None, None, None, None, None, None, None, None, None, None]
>>>
```